### PR TITLE
DPSv2 parser, SAS support and Unit Testing

### PR DESCRIPTION
**Pull request by CIPop**
Since DPSv2 is a copy of DPSv1, I recommend comparing the two versions files instead of using the PR diff to better understand the differences.
Most of the UT content has been copied and remains the same (except public API naming). There are changes to topics, API version and JSON responses.

### CODE REVIEW

File: sdk/tests/iot/provisioning_v2/test_az_iot_proviosining_v2_client_payload.c [hunk 1]
```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (RLeclair) [inline sdk/tests/iot/provisioning_v2/test_az_iot_proviosining_v2_client_payload.c]: I noticed that both the names of the v1 and v2 versions of this file spell provisioning wrong, not related to the current changes but might be good to address.

**Comment by RLeclair** [comment]
Noticed that this function: [az_iot_provisioning_v2_client.c Line 27](https://github.com/CIPop/azure-sdk-for-c/blob/v2_ut/sdk/src/azure/iot/az_iot_provisioning_v2_client.c#:~:text=AZ_INLINE%20az_span%20_az_iot_provisioning_v2_get_dps_v2_registrations_res()) has two "v2"s in the name. Is that intended?

**Review by RLeclair — APPROVED** [review]

File: sdk/tests/iot/provisioning_v2/test_az_iot_proviosining_v2_client_payload.c (hunk 1 above)
Comment (CIPop) [reply sdk/tests/iot/provisioning_v2/test_az_iot_proviosining_v2_client_payload.c]: Ouch... we didn't catch that in main. I will make the correction here.

**Comment by CIPop** [comment]
> Noticed that this function: [az_iot_provisioning_v2_client.c Line 27](https://github.com/CIPop/azure-sdk-for-c/blob/v2_ut/sdk/src/azure/iot/az_iot_provisioning_v2_client.c#:~:text=AZ_INLINE%20az_span%20_az_iot_provisioning_v2_get_dps_v2_registrations_res()) has two "v2"s in the name. Is that intended?

No, looks like it was missed the previous PR. I will add fixes.

**Pull request merged**